sftp/sftp.go

```diff
@@ -0,0 +1,97 @@
+package sftp
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"golang.org/x/crypto/ssh"
+)
+
+const (
+	// PasswordAuth indicates password-based authentication
+	PasswordAuth = "passwordAuth"
+	// KeyAuth indicates key-based authentication
+	KeyAuth = "keyAuth"
+)
+
+// SFTPManager interface abstracts the SFTP client
+type SFTPManager interface {
+	UploadFile(localFilePath, remoteDir string) error
+	DownloadFile(remoteFilePath, localDir string) error
+	DeleteFile(remoteFilePath string) error
+}
+
+// SFTPManagerImpl is a real implementation of SFTPManager
+type SFTPManagerImpl struct {
+	client SFTPClient
+}
+
+func NewSFTPManager(sshClient *ssh.Client) (SFTPManager, error) {
+	sftpClient, err := NewSFTPClient(sshClient)
+	if err != nil {
+		return nil, fmt.Errorf("cannot create SFTP client: %w", err)
+	}
+	return &SFTPManagerImpl{client: sftpClient}, nil
+}
+
+// UploadFile uploads a file to the remote server
+func (r *SFTPManagerImpl) UploadFile(localFilePath, remoteDir string) error {
+	localFile, err := os.Open(localFilePath)
+	if err != nil {
+		return err
+	}
+	defer localFile.Close()
+
+	remoteFileName := filepath.Join(remoteDir, filepath.Base(localFilePath))
+	remoteFile, err := r.client.Create(remoteFileName)
+	if err != nil {
+		return err
+	}
+	defer remoteFile.Close()
+
+	_, err = io.Copy(remoteFile, localFile)
+	if err != nil {
+		fmt.Printf("Error copying file: %v\n", err)
+		return err
+	}
+
+	fmt.Printf("Uploaded %s to %s\n", localFilePath, remoteDir)
```

Comment:
In another comment I questioned even the need of a logger. I'm not against it, I think it can be used if @Gauravudia finds it useful for his use case but we should keep in mind that these things can be easily be logged on the caller side. This would avoid us the logger dependency completely.

Reply:
Removed `Print` statements from the SFTP package.